Unix/omi_error/omierror.c

```diff
@@ -382,12 +382,15 @@ MI_INLINE const MI_Char *Errno_ToString(
     }
 # else /* defined(CONFIG_ENABLE_WCHAR) */
     {
-        *buffer = '\0';
-
-        // Intentionally disregarding return code
-        (void)strerror_r(OMI_Code, buffer, len) ;
-
+#if defined(macos) || ((_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && ! _GNU_SOURCE)
+        int ret = strerror_r(OMI_Code, buffer, len);
+        if (ret != 0)
+            *buffer = '\0';
         return buffer;
+#else
+        char *ret = strerror_r(OMI_Code, buffer, len);
```

Comment:
I'm not familiar with POSIX rules in specific versions.

I'm unsure if this is correct. I guess, if an error is returned, you have no way to return that, and that's why you're zeroing the buffer (386-389)? But then at 391-392, that's not the case. Seems like an inconsistency in your behavior.

Is this really what you want?